src/HealthGPS.Datastore/datamanager.cpp

```diff
@@ -1,33 +1,86 @@
 #include "datamanager.h"
 #include "HealthGPS.Core/math_util.h"
 #include "HealthGPS.Core/string_util.h"
+#include "HealthGPS/program_path.h"
 
 #include <fmt/color.h>
 #include <fstream>
+#include <jsoncons/json.hpp>
+#include <jsoncons_ext/jsonschema/jsonschema.hpp>
 #include <rapidcsv.h>
 
 #include <utility>
 
+namespace {
+using namespace jsoncons;
+
+json resolve_uri(const jsoncons::uri &uri, const std::filesystem::path &schema_directory) {
+    constexpr const char *url_prefix = "http://example.com/";
+
+    const auto &uri_str = uri.string();
+    if (!uri_str.starts_with(url_prefix)) {
+        throw std::runtime_error(fmt::format("Unable to load URL: {}", uri_str));
+    }
+
+    // Strip URL prefix and load file from local filesystem
+    const auto schema_path = schema_directory / std::filesystem::path{uri.path().substr(1)};
+    auto ifs = std::ifstream{schema_path};
+    if (!ifs) {
+        throw std::runtime_error("Failed to read schema file");
+    }
+
+    return json::parse(ifs);
+}
+
+void validate_index(const std::filesystem::path &schema_directory, std::ifstream &ifs) {
+    // **YUCK**: We have to read in the data with jsoncons here rather than reusing
+    // the nlohmann-json representation :-(
```

Comment:
I guess `jsoncons` cannot be used as a drop-in replacement of `nlohmann-json` right?

Reply:
I don't think so, though they are similar.

In theory, the schema library can work with other JSON libraries (it accepts a template parameter for JSON object type), but unfortunately `nlohmann-json`'s implementation doesn't seem to be compatible with this. I should probably open an issue upstream